### PR TITLE
Grid

### PR DESCRIPTION
**Pull request by taylorallred**
screen shot 2016-10-18 at 4 25 39 pm

### CODE REVIEW

File: src/axes/HorizontalGrid.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    this.renderGrid();
+  }
+
+  componentDidUpdate () {
+    this.renderGrid();
+  }
+
+  getClassNames () {
+    const className = [
+      'background-grid horizontal-grid',
```
Comment (inlineblock) [inline src/axes/HorizontalGrid.js]: Just use `horizontal-grid`

File: src/axes/HorizontalGrid.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    if (this.props.className) {
+      className.push(this.props.className);
+    }
+    return className.join(' ');
+  }
+
+  render () {
+    return (
+      <g className={ this.getClassNames() } ref={ n => this.node = n } />
```
Comment (inlineblock) [inline src/axes/HorizontalGrid.js]: `n => this.node = d3.select(n)`

File: src/axes/HorizontalGrid.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    );
+  }
+
+  renderGrid () {
+    const group = d3.select(this.node);
+    const grid = d3.axisLeft(this.props.yScale)
+                   .tickSize(-this.props.width + this.props.paddingLeft + this.props.paddingRight)
```
Comment (inlineblock) [inline src/axes/HorizontalGrid.js]: should we use `-this.props.areaWidth`

File: src/axes/HorizontalGrid.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                   .tickFormat('');
+
+    if (this.props.ticks) {
+      grid.ticks(this.props.ticks);
+    }
+
+    const x = this.props.paddingLeft;
+    const y = this.props.paddingTop;
+    group.attr('transform', `translate(${x},${y})`);
```
Comment (inlineblock) [inline src/axes/HorizontalGrid.js]: inline this in the renderer

File: src/axes/HorizontalGrid.js (hunk 2 above)
Comment (inlineblock) [inline src/axes/HorizontalGrid.js]: `transform={ `translate(${this.props.paddingLeft},${this.props.paddingTop})` }`

File: src/axes/VerticalGrid.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                   .tickSize(-this.props.height + this.props.paddingTop + this.props.paddingBottom)
+                   .tickFormat('');
+
+    if (this.props.ticks) {
+      grid.ticks(this.props.ticks);
+    }
+
+    const x = this.props.paddingLeft;
+    const y = this.props.height - this.props.paddingBottom;
+    group.attr('transform', `translate(${x},${y})`);
```
Comment (inlineblock) [inline src/axes/VerticalGrid.js]: use react to render this.

File: src/axes/HorizontalGrid.js (hunk 1 above)
Comment (taylorallred) [inline src/axes/HorizontalGrid.js]: done

File: src/axes/HorizontalGrid.js (hunk 3 above)
Comment (taylorallred) [inline src/axes/HorizontalGrid.js]: done

File: src/axes/HorizontalGrid.js (hunk 4 above)
Comment (taylorallred) [inline src/axes/HorizontalGrid.js]: done

File: src/axes/VerticalGrid.js (hunk 5 above)
Comment (taylorallred) [inline src/axes/VerticalGrid.js]: done

**Pull request merged**